core/che-core-metrics-core/src/main/java/org/eclipse/che/core/metrics/ApiResponseCounter.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.core.metrics;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.binder.MeterBinder;
+import javax.inject.Singleton;
+
+@Singleton
+/**
+ * Metrics
+ *
+ * @author Mykhailo Kuznietsov
+ */
+public class ApiResponseCounter implements MeterBinder {
+  protected Counter informationalResponseCounter;
+  protected Counter successResponseCounter;
+  protected Counter redirectResponseCounter;
+  protected Counter clientErrorResponseCounter;
+  protected Counter serverErrorResponseCounter;
+
+  @Override
+  public void bindTo(MeterRegistry registry) {
+    informationalResponseCounter =
```

Comment:
I'd personally prefer having a single metric - `che.server.api.response` with different tag values.
E.g. you never have a `che.server.api.response.success` with any other code than `2xx`. That makes the tag useless (unless we also use that tag in other areas, which I doubt makes much sense).

So instead of having 5 different metrics, each with a unique code tag, I'd make a single metric with 5 different code tag values. This also makes it possible to easily aggregate the responses with e.g. 2xx and 3xx status codes.

Reply:
This can be done by just changing the counters to have the same name.

Reply:
alright